File: _Enutrof_[__-1]🎧🌝💉☕/Day 37/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        end = -1
        count = 0
        for i, interval in enumerate(intervals):
            if i > 0:
                if interval[0] >= end:
                    end = interval[1]
                else:
                    count += 1
                    end = min(end, interval[1])
            else:
                end = interval[1]
        return count

File: Engr-T.stark/nonOverlappingIntervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int numRemovals = 0;
        sort(intervals.begin(), intervals.end());
        vector<int> prev = intervals[0];
        for(int i = 1; i < intervals.size(); i++) {
            if(intervals[i][0] < prev[1]) {
              numRemovals += 1;
                prev[1] = min(intervals[i][1], prev[1]);
            } 
            else {
                prev[1] = intervals[i][1]; 
            }
        }
        return numRemovals;
    }
};
